2199/2/participant_store.rb

```diff
@@ -0,0 +1,38 @@
+require_relative 'participant_name_loader'
+require_relative 'participant'
+require_relative 'battle'
+# class Participant store
+class ParticipantStore
+  def initialize(battles_folder)
+    @battles_folder = battles_folder
+    @participants = {}
+  end
+
+  def top_by_bad_words(limit)
+    participants.values.sort_by { |participant| - participant.bad_words_count }.first(limit)
+  end
+
+  private
+
+  def participants
+    if @participants.empty?
+      Dir.glob("#{@battles_folder}/*") do |filename|
+        next unless File.file?(filename)
+        add_participant_from_file(filename)
```

Comment:
Those 2 methods `#participants` and `#add_participant_from_file` look bad: they are long, the private method is called with arguments (usually means there are too many responsibilities for this class), temporary variable `@participants` is used, we mutate participants by calling `#add_battle` etc.

Let's rewrite this. I suggest to reduce a list of files to a hash like this:
```rb
{
  "First Battler Name" => ["First Battle Text", "Second Battle Text"],
  "Oxxxymiron" => ["Говно залупа пенис ...", "..."]
}
```

Then we can map this hash to the array of participants.
```rb
parsed_files.map do |name, battle_texts|
  Participant.new
end
```

But don't pass battles to participant one by one using `#add_battle`. Pass the array of battles to the initializer of a participant. No mutations.